mysql/mysql-lib.pl

```diff
@@ -1412,6 +1412,28 @@ sub list_collation_orders
 return sort { lc($a->[0]) cmp lc($b->[0]) } @rv;
 }
 
+# fix_collation(file)
+# Fixes unsupported collations on restore, by replacing
+# unsuported with the closest supported variant
+sub fix_collation
+{
+my ($file) = @_;
+my ($version, $variant) = &get_remote_mysql_variant();
+if ($variant eq 'mariadb') {
+	my $tfile = &transname();
+	open(IN, '<' . $file) or die $!;
```

Comment:
Is this safe to read and write to the same file? Seems like we should use a temp file ..

Reply:
> Is this safe to read and write to the same file? Seems like we should use a temp file ..

You're right. But this is what I already do .. ?

Reply:
Oh right I misread the filenames! Looks good ..